### PR TITLE
Adds middleman-vegas gem to handle syntax highlighting

### PR DESCRIPTION
**Pull request by burtlo**
Under the covers this uses the same rouge syntax lexer/highlighter
that was there before (albiet a later version). The gem provides
support to code fences and a code helper which is the same as
before but now it allows you to specify additional metadta after
the code language declaration for the code fence.

* Removes the javascript library that performed this highlighting
* Gemfile makes use of the new gem
* Config has been updated to load the gem by name
* A Rouge Lexer for Habitat Studios has been added to
  assist with processing the tokenization and ultimately
  its style and colorization
* Stylesheet has been added to support the new code format

When merged the following content (https://github.com/burtlo/middleman-vegas#syntax) with a few more concrete examples would be added to the wiki.

Signed-off-by: Franklin Webber <franklin@chef.io>

## Syntax

You can define code fences as you would normally. These code fences have support for the languages defined in [rouge](https://github.com/jneen/rouge/wiki/List-of-supported-languages-and-lexers).

Define a code fence with a language.

    ```lang
    [code]
    ```

Define a code fence for a language with the addition of a title.

    ```lang title
    [code]
    ```

Define a code fence for a language with additional metadata

    ```lang [metadata]
    [code]
    ```

The additional metadata that can be specified:

| Metadata     | Example                    | Description                                                           |
|:-------------|:---------------------------|:----------------------------------------------------------------------|
|`lang`        | `ruby`                     | Used by the syntax highlighter. Passing 'plain' disables highlighting.|
|`title`       | `title:"Figure 1.A"`       | Add a figcaption title to your code block. |
|`url`         | `url:"https://github.com"` | No default value |
|`link_text`   | `link_text:"Download"`     | Text for the link, default: `"link"` |
|`linenos`     | `linenos:true`             | Enables line numbering |
|`start`       | `start:5`                  | Start the line numbering at the given value. |
|`mark`        | `mark:1-4,8`               | Highlight lines of code. This example marks lines 1,2,3,4 and 8 |
|`class`       | `class:"css example"`      | Add CSS class names to the code `<figure>` element |

## Example Output

image

## Checkout the branch

```shell
$ git remote add burtlo https://github.com/burtlo/habitat.git
$ git fetch burtlo www/code-fences
$ bundle
$ bundle exec middleman
```

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by burtlo** [comment]
To get the full effect of the syntax highlighting about 103 files will need to be updated. During my evaluation I performed a bit of this work to ensure that the highlighter would function properly across every page of the site.

How would you all like to have all the changes across all the pages and partials be delivered?

File: www/source/stylesheets/vendor/foundation/components/_button-group.scss [hunk 1]
```diff
@@ -84,7 +84,7 @@ $buttongroup-expand-max: 6 !default;
         &:first-child:nth-last-child(#{$i}) {
           &, &:first-child:nth-last-child(#{$i}) ~ #{$selector} {
             display: inline-block;
-            @if #{$buttongroup-spacing} == '0' {
+            @if '#{$buttongroup-spacing} == "0"' {
```
Comment (burtlo) [inline www/source/stylesheets/vendor/foundation/components/_button-group.scss]: This was a warning that kept appearing when I started middleman. The fix here was mistakenly checked in with all this other code. I am happy to leave it or remove it.

**Comment by cnunciato** [comment]
@burtlo Besides just having a smaller PR, what would be the value of not submitting all of the changes at once, assuming they're all related? I'd kind of like to see how everything looks together, personally.

**Comment by davidwrede** [comment]
I'm 👍  on one big PR with all the changes. I've got some stuff going on in the node web guide, but I'll just rebase on whatever you've got.

**Comment by burtlo** [comment]
@cnunciato and @davidwrede I can definitely do it as a big request by late tomorrow night. I definitely agree it would be good to see the entire thing together. I guess I didn't want to explore all of your minds with a 110 file changes kind of pull-request without a little a bit of an introduction.

**Comment by burtlo** [comment]
image

I need to update the gem to support indented code fences. Quite a few partials/documents use that approach.

**Comment by burtlo** [comment]
Alright I have pushed all the changes and adjusted every bit of code that I could find with the code helper or code fences. @cnunciato and @davidwrede if you want to give it a look.

**Comment by burtlo** [comment]
I was going to publish a hart file with these changes but I had more trouble then I thought I would getting the www site to build with scaffolding.

## Adventures in getting WWW into a Habitat Package

* The app logs locally to disk when launched by the `config.ru`. Updated the log to read an environment variable if present; otherwise, default. Defined an environment variable in the plan.

SUCCESS!

* Next, eventmachine was causing issues and I thought I tracked it down to simply needing openssl. Of course, I didn't stare at the [source](https://github.com/eventmachine/eventmachine/blob/master/ext/extconf.rb) for long to see what it was that it may have needed.

FAIL

* eventmachine was tied to livereload. I didn't plan on needing live reload when run by a Supervisor so I disabled that feature and gem for my experiment. Ideally I would learn what gem group is ignored by scaffolding-ruby.

* Then I found that `middleman-core` uses listen to watch for file changes. Which made me think I could pass something like `bundle exec middleman server --watcher-disable`. But then looking at the `config.ru` we use the middleman class app way to boot the site. Their main docs don't point to any good examples. I'm sure if I dug into the source I could fine the options.

* Thought about building the static site and throwing it behind nginx or apache. Ran `middleman build` and worked two bugs out with some of the templating updates I had made.

* Next step would be learn more nginx / apache to drop in the site but I'm off to bed.

**Comment by eeyun** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/298073385).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by burtlo** [comment]
To view acceptance go to https://www.acceptance.habitat.sh/

**Comment by cnunciato** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/311484314).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/311484314).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by cnunciato** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/311498072).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/311498072).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**